heat/core/linalg/solver.py

```diff
@@ -269,3 +270,196 @@ def lanczos(
         V.resplit_(axis=None)
 
     return V, T
+
+
+def solve_triangular(A: DNDarray, b: DNDarray) -> DNDarray:
+    """
+    This function provides a solver for (possibly batched) upper triangular systems of linear equations: it returns x in Ax = b, where A is a (possibly batched) upper triangular matrix and
+    b a (possibly batched) vector or matrix of suitable shape, both provided as input to the function.
+    The implementation builts on the corresponding solver in PyTorch and implements a block-wise version thereof.
+
+    Parameters
+    ----------
+    A : DNDarray
+        An upper triangular (possibly batched) invertible square (n x n) matrix, i.e. an DNDarray of shape (..., n, n).
+    b : DNDarray
+        a (possibly batched) n x k matrix, i.e. an DNDarray of shape (..., n, k), where the batch-dimensions denoted by ... need to coincide with those of A.
+        Vectors have to be provided as n x 1 matrices and the split dimension of b must the second last dimension if not None.
```

Comment:
Not sure I understand this sentence (line 287), esp. in relation to the previous one.

Reply:
Has been (hopefully) clarified.